src/swap/SwapReviewScreen.tsx

```diff
@@ -59,8 +60,39 @@ export function SwapReviewScreen() {
   const [swapFeeModalVisible, setSwapFeeModalVisible] = useState(false)
   const [swapResponse, setSwapResponse] = useState<ZeroExResponse | null>(null)
   const [fetchError, setFetchError] = useState(false)
-  const allTokens = useSelector(tokensListSelector)
+  const tokensByAddress = useSelector(tokensByAddressSelector)
   const walletAddress = useSelector(walletAddressSelector)
+  const celoAddress = useSelector(celoAddressSelector)
+  const feeCurrency = useFeeCurrency() ?? celoAddress
+
+  const estimateFeeAmount = () => {
+    if (!feeCurrency || !swapResponse || !celoAddress || !tokensByAddress) {
+      return new BigNumber(0)
+    }
+
+    const celoUsdPrice = tokensByAddress[celoAddress]?.usdPrice
+    const feeCurrencyUsdPrice = tokensByAddress[feeCurrency]?.usdPrice
+
+    if (!celoUsdPrice || !feeCurrencyUsdPrice) {
+      return new BigNumber(0)
+    }
+
+    const estimatedCeloFeeAmount = divideByWei(
+      new BigNumber(swapResponse.unvalidatedSwapTransaction.gas).multipliedBy(
+        new BigNumber(swapResponse.unvalidatedSwapTransaction.gasPrice)
+      )
```

Comment:
nit: I think we can do this:
```suggestion
      new BigNumber(swapResponse.unvalidatedSwapTransaction.gas).multipliedBy(
        swapResponse.unvalidatedSwapTransaction.gasPrice
      )
```